otelcolbuilder/cmd/configprovider.go

```diff
@@ -34,9 +36,18 @@ import (
 // https://github.com/open-telemetry/opentelemetry-collector/blob/65dfc325d974be8ebb7c170b90c6646f9eaef27b/service/command.go#L38
 
 func UseCustomConfigProvider(params *service.CollectorSettings) error {
+	// feature flags, which are enabled by default in our distro
+	err := featuregate.GetRegistry().Apply(map[string]bool{
+		"receiver.apache.emitServerNameAsResourceAttribute": true,
+		"receiver.apache.emitPortAsResourceAttribute":       true,
+	})
+
+	if err != nil {
+		return fmt.Errorf("setting feature gate flags failed: %s", err)
+	}
+
```

Comment:
Is it okay to return an error here instead of continuing? Afaik there's no logger in this function, so log and continue wouldn't work.

Reply:
This will cause otel to terminate with this error, which I think is fine, as this can only error if we try to set flags which don't exist, afaik.